src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        // Config values pulled from CoreCLR
+        private static readonly HillClimbing s_threadPoolHillClimber = new HillClimbing(4, 20, 100 / 100.0, 8, 15 / 100.0, 300 / 100.0, 4, 20, 10, 200, 1 / 100.0, 200 / 100.0, 15 / 100.0);
+
+        private class HillClimbing
+        {
+            public enum State
```

Comment:
The naming feels a bit odd because some of these are states of hill climbing, and some are reasons for changing the thread count (one-time transitions, not a state). Maybe call this `StateOrTransition` or something like that? Any call to `ChangeThreadCount` should be given a transition value. May also split the states from the transitions into separate enums.